libtransmission/variant.h

```diff
@@ -172,7 +172,14 @@ struct tr_variant
     };
 
     constexpr tr_variant() noexcept = default;
+#if __apple_build_version__ && __apple_build_version__ <= 11000033
+    // old clang incompatibility in Xcode 11.3.1 of `std::pair` with `delete`
+    tr_variant(tr_variant const&)
+    {
+    }
+#else
```

Comment:
We shouldn't be reaching the copy operator for `tr_variant` in the first place, since we try to never copy it by value.

Even if we did, we wouldn't want to leave it unimplemented :thinking: 

Where is this build error coming from?

Reply:
how to implement it?

Reply:
>We shouldn't be reaching the copy operator

We never reach it. Remember: it's a bug of an old compiler that we're trying to workaround. I put a break point: that copy is never happening.

>Where is this build error coming from?

Well, that buggy compiler won't say which file or which line. It's just saying that somewhere in the project:
```
In instantiation of member function 'std::__1::vector<std::__1::pair<unsigned long, tr_variant>, std::__1::allocator<std::__1::pair<unsigned long, tr_variant> > >::vector' requested here

Copy constructor of 'pair<unsigned long, tr_variant>' is implicitly deleted because field 'second' has a deleted copy constructor
```

And the very first commit to cause the build issue was #6575. So we're looking for a std::vector<std::pair<unsigned long, tr_variant>> in that pull request, and that is likely...
```
    using field_value_type = std::pair<const field_key_type, field_mapped_type>;
    using Fields = std::vector<field_value_type>;
```